lib/classes/PluginManager.js

```diff
@@ -309,6 +310,53 @@ class PluginManager {
     });
   }
 
+  loadVariableResolvers(pluginInstance) {
+    const pluginName = pluginInstance.constructor.name;
+    for (const [variablePrefix, resolverOrOptions] of _.entries(
+      pluginInstance.variableResolvers || {}
+    )) {
+      let options = resolverOrOptions;
+      if (!options) {
+        this.serverless.cli.log(
+          `Warning! Ignoring falsey variableResolver for ${variablePrefix} in ${pluginName}.`
+        );
+        continue;
+      }
+      if (_.isFunction(resolverOrOptions)) {
+        options = {
+          resolver: resolverOrOptions,
+        };
+      }
+      if (!_.isObject(options)) {
```

Comment:
I think defining JSON schemas (and handling validation with large tools as `ajv`)  would be an overkill for this case.

Reply:
Hmm, Yeah, does seem a little overkill. If we were already using that, i'd definitly do it.

Reply:
I don't disagree.  :+1:

It may be good to introduce at some point, I know that in other projects I've seen it nicely reduce and clarify the code base.

In this case it seemed like it might support clear communication with the plug-in developers that will use the API.  This said I don't think it's terribly important.